src/common/dependency-manager/local-manager.ts

```diff
@@ -95,7 +95,23 @@ export default class LocalDependencyManager extends DependencyManager {
         }
         component_config.setService(service_ref, service);
       }
+
+      for (const [param_key, param_value] of Object.entries(component_config.getParameters())) {
```

Comment:
Do we also need to make these changes when registering a component? 
If so we should probably add tests for register that checks the posted body on the register call.

Makes me think that a purely regex replacement might be better as opposed to spot changes. Potentially would allow us to use the same code for register/deploy. I know we had talked about a regex approach did you rule that out?

Reply:
I liked the idea of it because I think adding the file ref replacement in the builders because can prevent a lot of problems down the line related to relative file paths. I was pretty concerned about it though because of the weird ways that it could go wrong. Hard to say that the regexes for json and yaml are bulletproof, but I added a bunch of tests to be more sure, and will probably add more